### ISSUE TITLE
<Bitstamp> Error with the credentials.

### ISSUE THREAD

**Issue by imjonathan** [butor/blackbird#171]
<Bitstamp> Error with the credentials.

Despite supplying valid info I get the <Bitstamp> Error with the credentials in the top of the log file...

[ Current balances ]
<Bitstamp> Error with the credentials.
<Bitstamp> Error with the credentials.
   Bitstamp:    0.00 USD        0.000000 BTC

Initially I set up a sub account on bitstamp and created an API key.

I set the following with the supplied Key and Secret:
BitstampApiKey=KEY_NOT_SHOWN
BitstampSecretKey=SECRET_NOT_SHOWN

It it not clear if the BitstampClientId should be set to the "Customer ID" as supplied by BitStamp or the "Sub account" number (seen as far as I can tell only on the api key page). In any case I tried both. I confirmed the the API key had "Account balance" access (in fact I gave it trade access as well)

I confirmed that I set:
DemoMode=false

I tried quoting the keys and secrets with nothing, with single and with double quotes.

I repeated the process with a new key.

When all these failed I tried making an API key for the main account and I got the same result.

No matter what I always get <Bitstamp> Error with the credentials.

Any thoughts?

**Comment by greatwolf** [comment]
I don't have an apikey setup on stamp so it'll be pretty difficult for me to test. (AML/KYC such a pain)

If I had to make a guess, could the datatype for the balance returned be wrong? It'll be helpful to see what the actual json response was. To do that add the following line and recompile:

    *params.logFile << "<Bitstamp> Error with JSON: " << json_dumps(root, 0);

in https://github.com/butor/blackbird/blob/master/src/exchange/bitstamp.cpp#L67.

**Comment by imjonathan** [comment]
Thanks so much for the quick reply.

I inserted that before line 67 and now the error reads:
<Bitstamp> Error with JSON: {"error": "API key not found"}<Bitstamp> Error with the credentials.
<Bitstamp> Error with JSON: {"error": "API key not found"}<Bitstamp> Error with the credentials.

**Comment by greatwolf** [comment]
That is suggesting something's wrong with your api key, https://www.bitstamp.net/api/#authentication-errors. Are you sure it's correct?

**Comment by imjonathan** [comment]
I am 100% sure these are correct (I've tried 4 separate times with different keys):
BitstampApiKey=
BitstampSecretKey=

I am not sure this is correct (but I have tried all conceivable possibilities).
BitstampClientId=
 
For Client ID 
I have tried the value found here: https://www.bitstamp.net/account/balance/ (as suggested in the API AUTHENTICATION -> SIGNATURE section of https://www.bitstamp.net/api/ )

Mine has a leading zero, I have also tried omitting that.

I have also tried the id value for the sub account shown on the API key page.
And I tried not using a sub account...
